client/a8c-for-agencies/components/items-dashboard/items-dataviews/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { Spinner } from '@automattic/components';
+import { DataViews } from '@wordpress/dataviews';
+import { useTranslate } from 'i18n-calypso';
+import { ReactNode } from 'react';
+import ReactDOM from 'react-dom';
+import { ItemsDataViewsType, DataViewsColumn } from './interfaces';
+// todo: Extract from style.scss not common styles (colors and specific to Jetpack Cloud components)
+import './style.scss';
+
+const getIdByPath = ( item: object, path: string ) => {
+	const fields = path.split( '.' );
+	let result: any = item;
+	for ( const field of fields ) {
+		if ( result[ field ] === undefined ) {
+			return undefined;
+		}
+		result = result[ field ];
+	}
+	return result;
+};
+
+/**
+ * Create an item column for the DataViews component
+ * @param id
+ * @param header
+ * @param displayField
+ * @param getValue
+ * @param isSortable
+ * @param canHide
+ */
+export const createItemColumn = (
+	id: string,
+	header: ReactNode,
+	displayField: () => ReactNode,
+	getValue: () => undefined,
+	isSortable: boolean = false,
+	canHide: boolean = false
+): DataViewsColumn => {
+	return {
+		id,
+		enableSorting: isSortable,
+		enableHiding: canHide,
+		getValue,
+		header,
+		render: displayField,
+	};
+};
+
+export type ItemsDataViewsProps = {
+	data: ItemsDataViewsType< any >;
+	isLoading?: boolean;
+	// todo: is it necessary? Could we get it in this component?
```

Comment:
> Is it isLargeScreen or isLoading?

isLargeScreen

> If isLargeScreen, since it's used a lot in JetpackSitesDataViews it makes sense to pass it through as a prop to ItemsDataViews instead of re-creating it - if we were using it in ItemsDataViews. I don't see that we are, nor that it is passed as a prop to DataViews. So it looks like it isn't necessary.

Perfect! Thank you for the code comment clarification. I'll remove it from ItemsDataViews.